### ISSUE TITLE
[Carbon X] DatePicker color tokens applied with g90 and g100 themes have inaccessible color contrast

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/carbon#1710]
## Detailed description

A number of colors/tokens being applied to the DatePicker are not accessible when the `g100` or `g90` themes from `@carbon/themes` are used.

Here are some example screenshots...

#### Simple date picker
<img width="305" alt="screen shot 2019-01-29 at 4 40 08 pm" src="https://user-images.githubusercontent.com/9057921/51994655-e2b89800-2476-11e9-9ac6-a4f0d5522994.png">

#### Range with min and max
<img width="315" alt="screen shot 2019-01-30 at 10 06 47 am" src="https://user-images.githubusercontent.com/9057921/51994653-e2b89800-2476-11e9-86c5-56ecf8092629.png">

#### Range
<img width="304" alt="screen shot 2019-01-30 at 10 01 51 am" src="https://user-images.githubusercontent.com/9057921/51994654-e2b89800-2476-11e9-80e4-a2b1df04b268.png">

**Comment by jendowns** [comment]
So this isn't a blocker for us right now, since I can override colors selectively in our addons library to make them accessible.

However, I really want to make sure I have the right intended color for the bottom border on the input element -- just so my override is in the right ballpark, at least. 🤔 

@IBM/carbon-designers could anyone let me know what color that bottom input border should be for a g100 theme? Thank you 🙏  (EDIT: I'm referring to the bottom border color as shown below, before the input is select/focused/active)

<img width="319" alt="screen shot 2019-01-30 at 11 27 02 am" src="https://user-images.githubusercontent.com/9057921/52000148-fc131180-2481-11e9-8149-80b4f50ea6a2.png">

**Comment by aagonzales** [comment]
- The underline for the fields should be `$ui-04` which is Gray 60 `#6F6F6F` in the Gray 100 theme
- All the icons should be using the token `$icon-01`
- The day ranges are just hard coded and will have to be over ridden. It seems excessive to make a token just for that. Could be convinced otherwise though. Could maybe be like a `$highlight` token or something, could maybe be used elsewhere
- The placeholder text in the fields should be using `$text-03` which should be Gray 60 `#6F6F6F` in the G100 Theme

**Comment by aagonzales** [comment]
What the Duo file says the dark date picker should look like (ignore their field styles): 
- highlight is Blue 90 `#021677`
<img width="365" alt="image" src="https://user-images.githubusercontent.com/11670886/52139633-4e853700-2616-11e9-8566-fb9eededfe19.png">

**Comment by jendowns** [comment]
I'm waiting for https://github.com/IBM/carbon-components/pull/1795 to be merged (as I need the new `$highlight` token from `@carbon/themes`) before I can proceed with proposing a fix for this 👍

**Comment by joshblack** [comment]
@jendowns just a heads up, will pull out the updates into a two parts, I think. Hoping to have it done this morning if folks have a chance to leave feedback 👍

**Comment by jendowns** [comment]
Sounds good -- thank you!

**Comment by jendowns** [comment]
@aagonzales Quick question -- That "selected" color (the start/end selected date in a range) and the text + "dot" color for the current date look like blue60 in these dark themes. The token used throughout looks like `$interactive-01`, which unfortunately is always blue60 for all themes -- https://carbon-elements.netlify.com/themes/examples/preview/

Am I correct that the blue for select/today should be blue40 (or even blue50) instead? Do you think `$interactive-01` needs to be modified... or a new token needs to be created?

**Comment by aagonzales** [comment]
@jendowns we're one step a head of you this time. We just added in a new `$interactive-04` that will those lighter blues in the dark themes but still keeping Blue 60 in the dark themes for the primary buttons with `$interactive-01` https://github.com/IBM/carbon-elements/issues/366

**Comment by aagonzales** [comment]
I think this has been resolved with the addition of the `$highlight` and `$interactive-04` color tokens.
